### ISSUE TITLE
How to impove conv2d performance in cpu mode 

### ISSUE THREAD

**Issue by daydayfun** [pytorch/pytorch#35153]
### Please note that this issue tracker is not a help form and this issue will be closed.

We have a set of [listed resources available on the website](https://pytorch.org/resources). Our primary means of support is our discussion forum:

- [Discussion Forum](https://discuss.pytorch.org/)

1. when i use PIP to install pytorch, con2d perforamance is better

2. when i down pytorch 1.0.0 source code in gitlab, bad performance

3.USE_MKL or OPENMP or some other optimizition results in the performance difference?

Here is  con2d op shape:
Conv2d(64, 3, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
way 1:
pytorch install:
conda install pytorch-cpu==1.0.0 torchvision-cpu==0.2.1 cpuonly -c pytorch
log: the conv2D infer time :3.85 s
way 2:
down pytorch 1.0.0 source code and compile with "python setup.py install"
log: the conv2D infer time :13.63 s

**Comment by daydayfun** [comment]
import torch
from torch.nn.parameter import Parameter
import torch.nn as nn
from torch.nn import functional as F
import time


in_shape = (1,64,1440,2560)
weight_shape = (3,64,3,3)
bias_shape = (3)
inputs = torch.rand(in_shape, dtype=torch.float)
fliters = torch.rand(weight_shape, dtype=torch.float)
bias = torch.ones(bias_shape, dtype=torch.float)

stride = 1
padding = [1,1]
dilation = 1
groups = 1
t1 = time.time()
out_cpu = F.conv2d(inputs, fliters, bias = bias, stride = stride, padding = padding, dilation = dilation, groups = groups)
t2 =time.time()
print('conv2d time', t2-t1);

**Comment by daydayfun** [comment]
I can find out "install pytorch-cpu==1.0.0 torchvision-cpu==0.2.1 cpuonly -c pytorch"  how to compile the pytorch code, whether some optimization enableed?  Please help me with this, thanks a lot.

**Comment by daydayfun** [comment]
I can't  find out "install pytorch-cpu==1.0.0 torchvision-cpu==0.2.1 cpuonly -c pytorch" how to compile the pytorch code, whether some optimization enableed? Please help me with this, thanks a lot.

**Comment by daydayfun** [comment]
static void THNN_(SpatialConvolutionMM_updateOutput_frame)(
          THTensor *input,
          THTensor *output,
          THTensor *weight,
          THTensor *bias,
          THTensor *finput,
          int kW,
          int kH,
          int dW,
          int dH,
          int padW,
          int padH,
          int64_t nInputPlane,
          int64_t inputWidth,
          int64_t inputHeight,
          int64_t nOutputPlane,
          int64_t outputWidth,
          int64_t outputHeight)
{
  int64_t i;
  THTensor *output2d;

  THNN_(unfolded_copy)(finput, input, kW, kH, dW, dH, padW, padH,
		       nInputPlane, inputWidth, inputHeight,
		       outputWidth, outputHeight);
**//unfolded_copy -->cost  6123 ms**
  output2d = THTensor_(newWithStorage2d)(THTensor_getStoragePtr(output), output->storage_offset(),
                                         nOutputPlane, -1,
                                         outputHeight*outputWidth, -1);
  if (bias) {
    for(i = 0; i < nOutputPlane; i++)
        THVector_(fill)
	  (THStorage_(data)(THTensor_getStoragePtr(output)) + output->storage_offset() + output->stride(0) * i,
	   THTensor_(get1d)(bias, i), outputHeight*outputWidth);
  } else {
    THTensor_(zero)(output);
  }
**//addmm cost 7753ms**
  THTensor_(addmm)(output2d, 1, output2d, 1, weight, finput);

  c10::raw::intrusive_ptr::decref(output2d);
}

**Comment by daydayfun** [comment]
THTensor_(addmm) 
THNN_(unfolded_copy)
cost  a lot of time, how to optimize them??

**Comment by ptrblck** [comment]
The PyTorch binaries ship with some additional libraries, which accelerate your use cases.
E.g. MKL might speed up your convolutions, so you would need to install this lib when building PyTorch from source.

That being said, I would recommend to stick to the lastest stable release, as `1.0.0` is old by now.

**Comment by daydayfun** [comment]
Thanks very much，i will install mkl-dnn and try
I found the difference between libtorch.so, the faster one has mkl-dnn inside.
